R/fill_ch_names.R

```diff
@@ -43,187 +58,721 @@ fill_ch_names <- function(ch_data,
+  # then we accept ch_name_lookup and overwrite ch_name
+
+  col_to_select <- c(
+    "unique_identifier",
+    "matching_quality_indicator_overall",
+    "sending_location",
+    "latest_sc_id",
+    "chi",
+    "ch_name",
+    "ch_postcode",
+    "social_care_id",
+    "period",
+    "period_start_date",
+    "period_end_date",
+    "ch_provider",
+    "reason_for_admission",
+    "type_of_admission",
+    "nursing_care_provision",
+    "ch_admission_date",
+    "ch_discharge_date",
+    "age",
+    "record_date",
+    "qtr_start",
+    "latest_flag",
+    "gender",
+    "dob",
+    "postcode",
+    "date_of_death",
+    "ch_name_validated",
+    "open_interval",
+    "latest_close_date",
+    "ch_name_old",
+    "ch_postcode_old",
+    "ch_name_keyword"
+  )
+
+  ch_pc_match <- ch_pc_match %>%
+    dplyr::mutate(matching_quality_indicator_overall = dplyr::if_else(.data[["matching_quality_indicator_overall"]] == 100L &
+      .data[["postcode_matching"]],
+    14L,
+    .data[["matching_quality_indicator_overall"]]
+    )) %>%
+    # now remove cases of quality being 100L for the next section:
+    # ch_name matching
+    dplyr::filter(.data[["matching_quality_indicator_overall"]] != 100L) %>%
+    dplyr::mutate(
+      ch_name_old = .data[["ch_name"]],
+      ch_postcode_old = .data[["ch_postcode"]],
+      ch_name = .data[["ch_name_validated"]],
+      ch_postcode = .data[["ch_postcode_lookup"]]
+    ) %>%
+    dplyr::select(dplyr::all_of(col_to_select))
+
+  ## matching by ch_name, quality 15L-21L ----
+  ### perfect matching by ch_name, and main part of postcode, quality 15L ----
+  # ch_name matching, then overwrite postcode from ch_lookup
+  # 15L means perfect matching name,
+  # and relevant dates align,
+  # but not the main part of the postcode, say "EH12"
+
+  ch_name_match1 <- ch_data %>%
+    dplyr::anti_join(ch_pc_match,
+      by = dplyr::join_by("unique_identifier")
     ) %>%
-    dplyr::filter(
-      !is.na(.data[["ch_name"]]) & !is.na(.data[["ch_postcode"]])
+    dplyr::inner_join(
+      ch_name_lookup,
+      by = dplyr::join_by(
+        x$ch_name == y$ch_name_validated,
+        x$ch_admission_date <= y$latest_close_date,
+        x$ch_admission_date >= y$ch_date_registered,
+        # some care homes have same name, so use ch_pc_partial2 to filter
+        "ch_pc_partial3"
+      )
+    ) %>%
+    dplyr::mutate(
+      ch_name_old = .data[["ch_name"]],
+      ch_postcode_old = .data[["ch_postcode"]],
+      ch_name_validated = .data[["ch_name"]],
+      # ch_name_validated is omitted because of join_by(), add back
+      ch_postcode = .data[["ch_postcode_lookup"]],
+      matching_quality_indicator_overall = 15L
     ) %>%
-    dplyr::left_join(ch_name_best_match,
+    dplyr::select(dplyr::all_of(col_to_select))
+
+  ### fizzy matching by ch_name, and matching main part of postcode, quality 16L ----
+  ch_match <- dplyr::bind_rows(ch_pc_match, ch_name_match1)
+
+  ch_name_match2 <- ch_data %>%
```

Comment:
Would i be possible to rename the dataframes with what quality indicator they are being assigned? so like ch_name_match_indicator16, and ch_pc_match_indicator1to12 or something.

Reply:
Variables' names changed